kumascript/index.js

```diff
@@ -13,7 +13,24 @@ const {
 } = require("./src/live-sample.js");
 const { HTMLTool } = require("./src/api/util.js");
 
-const renderFromURL = memoize(async (url) => {
+const DEPENDENCY_LOOP_INTRO =
+  'The following documents form a circular dependency when rendering (via the "page" and/or "IncludeSubnav" macros):';
+
+const renderCache = new Map();
+
+const renderFromURL = async (url, urlsSeen = null) => {
+  const urlLC = url.toLowerCase();
```

Comment:
This just feels a bit weird. But it's necessary. 
The `renderFromURL` is given a URL that it needs to not mess with because it contains important information about the *caSe* of the URL. That URL is used to build live samples iframe `src` attributes for example. If you lowercase it (just for the benefit of the cache and the `urlsSeen`) you're changing it and it breaks how it gets used within the live sample macro. 

However, we should NOT be using `info.cleanURL`, like we do for prerequisites, because that one might use redirects which only makes sense when you're rendering macros. The original caller of this function comes ultimately from the CLI that builds all pages and there it can't be a redirect'ing URL.